test/qpy_compat/test_qpy.py

```diff
@@ -973,6 +1003,30 @@ def load_qpy(qpy_files, version_parts):
                 circuit, qpy_circuits[i], i, version_parts, bind=bind, equivalent=equivalent
             )
 
+    while pulse_files:
+        path, version = pulse_files.popitem()
+
+        if version_parts < version or version_parts >= (2, 0):
+            continue
+
+        if path == "pulse_gates.qpy":
+            try:
+                load(open(path, "rb"))
```

Comment:
Do we need to use a context manager here to avoid forcing python to close the file when the file object is dropped? I think at one point python was emitting warnings about that.

```suggestion
                with open(path, "rb") as fd:
                    load(fd)
```

Reply:
Agreed, although it's only a handful of files opened for reading but it's better practice anyway with a context manager.